includes/class-amp-theme-support.php

```diff
@@ -1497,6 +1497,57 @@ protected static function has_dependency( WP_Dependencies $dependencies, $curren
 		return false;
 	}
 
+	/**
+	 * Check if a handle is exclusively a dependency of another handle.
```

Comment:
Before reviewing the actual logic, let me try to rephrase here to make sure I understand the intent and we're on the same page here.

The way I understand it is that this method checks whether a dependency is *only* being pulled in as a child dependency of a given parent dependency and nowhere else (neither as a child dependency of another parent dependency, nor as a main dependency itself).

This is meant to be used to detect dependencies that are either safe to remove when their parent was removed or safe to add to devmode if their parent was added to devmode.

Is the above correct?

Reply:
> (neither as a child dependency of another parent dependency

I don't think this part is quite right. It should be allowed to a (recursive) dependency of something else, if that thing is also ultimately a dependency of the dependency in question.

> This is meant to be used to detect dependencies that are either safe to remove when their parent was removed or safe to add to devmode if their parent was added to devmode.

The latter, yes, whether we can add dev mode to the dependency. We only want to add dev mode to dependences which are ultimately and exclusively _connected_ with the admin bar. So this is what this method is trying to check for.